src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -223,14 +221,22 @@ public static class PrereleaseCompatibilityTransformer
 
                     JObject tool = (JObject)run["tool"];
                     modifiedLog |= RenameProperty(tool, previousName: "fileVersion", newName: "dottedQuadFileVersion");
-                    if (tool != null && tool["language"] == null) { tool["language"] = "en-US"; }
+                    if (tool != null && tool["language"] == null)
+                    {
+                        tool["language"] = "en-US";
+                        modifiedLog = true;
+                    }
 
                     JObject conversion = (JObject)run["conversion"];
                     if (conversion != null)
```

Comment:
> if (conversion != null) [](start = 19, length = 24)

this code and above could be DRY'ed out. i.e., author a little method PopulateValueIfNotAlreadyPresent(tool, "language", "en-us");

Reply:
ok.

---
In reply to: [248881391](https://github.com/Microsoft/sarif-sdk/pull/1221#discussion_r248881391) [](ancestors = 248881391)